nautobot/extras/signals.py

```diff
@@ -104,14 +105,14 @@ def handle_cf_removed_obj_types(instance, action, pk_set, **kwargs):
     Handle the cleanup of old custom field data when a CustomField is removed from one or more ContentTypes.
     """
     if action == "post_remove":
-        instance.remove_stale_data(ContentType.objects.filter(pk__in=pk_set))
+        delete_custom_field_data.delay(instance.name, pk_set)
```

Comment:
Huh. Given that we have `CASCADE` deletion on `CustomField -> CustomFieldChoice`, can we successfully delete a `TYPE_SELECT` `CustomField` that is in use on various models, or do the prohibitions against deleting an in-use `CustomFieldChoice` get in the way? I *think* it should work but it'd be great to see explicit testing around this.

Reply:
Yes, this works because Django's `CASCADE` ignores the subordinate's `delete()` method. I added some tests though.